mirrord/sip/src/lib.rs

```diff
@@ -1,3 +1,31 @@
+//! Concerns MacOS' SIP (System Integrity Protection) mechanism and how to sidestep it.
+//!
+//! ### Why?
+//!
+//! SIP makes it such that the `mirrord-layer` lib cannot be dynamically loaded into the user's
```

Comment:
We could mention it does that by just removing the relevant env vars when starting the process.

Reply:
Also, if we want to really tell the whole truth - it's not always SIP that we need to bypass - it's sometimes the hardened runtime.